src/mesh/grids/non_periodic_cube.cpp

```diff
@@ -0,0 +1,66 @@
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <Sacado.hpp>
+#include "non_periodic_cube.h"
+
+namespace PHiLiP {
+namespace Grids {
+
+template<int dim, typename TriangulationType>
+void non_periodic_cube_flow(
+    TriangulationType& grid,
+    double domain_left,
+    double domain_right,
+    bool colorize,
+    bool shock_tube,
+    bool shu_osher) 
+{
+    dealii::GridGenerator::hyper_cube(grid, domain_left, domain_right, colorize);
+
+    if (shock_tube || shu_osher) {
+        for (auto cell = grid.begin_active(); cell != grid.end(); ++cell) {
+            // Set a dummy boundary ID
+            cell->set_material_id(9002);
+            if (cell == grid.begin_active())
+            {
+                for (unsigned int face = 0; face < dealii::GeometryInfo<dim>::faces_per_cell; ++face) {
+                    if (shu_osher) {
+                        // Set left boundary to Riemann Far Field condition
```

Comment:
Since it is 1D, I thought it would be alright to loop through the faces since there isn't a top or bottom boundary and only set the ones on the boundary to an ID. I've changed this now since the three cases I'm running only have one cell. Instead of a for loop, I am just explicitly setting face(0) and face(1) to a boundary ID depending on the test case being run.